libraries/AP_BattMonitor/AP_BattMonitor_SMBus.cpp

```diff
@@ -46,71 +46,58 @@ void AP_BattMonitor_SMBus::read(void)
 }
 
 // reads the pack full charge capacity
-// returns true if the read was successful, or if we already knew the pack capacity
-bool AP_BattMonitor_SMBus::read_full_charge_capacity(void)
+// returns if we already knew the pack capacity
+void AP_BattMonitor_SMBus::read_full_charge_capacity(void)
 {
-    uint16_t data;
-
     if (_full_charge_capacity != 0) {
-        return true;
-    } else if (read_word(BATTMONITOR_SMBUS_FULL_CHARGE_CAPACITY, data)) {
-        _full_charge_capacity = data * get_capacity_scaler();
-        return true;
+        return;
     }
-    return false;
+    read_word(BATTMONITOR_SMBUS_FULL_CHARGE_CAPACITY, _full_charge_capacity);
```

Comment:
let's be pedantic here and keep checking the read_word return, that is cheaper than the *= anyway